datasets/titanic/metadata.json

```diff
@@ -1,30 +1,70 @@
 {
     "@context": {
-      "@vocab": "https://schema.org/",
-      "csvw": "http://www.w3.org/ns/csvw",
-      "ml": "http://mlcommons.org/schema/"
+      "sc": "https://schema.org/",
+      "ml": "http://mlcommons.org/schema/",
+      "wd": "https://www.wikidata.org/wiki/Q"
     },
-  
-    "@type": "Dataset",
+
+    "@type": "sc:Dataset",
     "@language": "en",
     "name": "Titanic",
     "url": "https://www.openml.org/d/40945",
     "description": "The original Titanic dataset, describing the status of individual passengers on the Titanic.\n\n The titanic data does not contain information from the crew, but it does contain actual ages of half of the passengers. \n\n For more information about how this dataset was constructed: \nhttps://web.archive.org/web/20200802155940/http://biostat.mc.vanderbilt.edu/wiki/pub/Main/DataSets/titanic3info.txt\n\nOther useful information (useful for prices description for example):\nhttp://campus.lakeforest.edu/frank/FILES/MLFfiles/Bio150/Titanic/TitanicMETA.pdf\n\n Also see the following article describing shortcomings of the dataset data:\nhttps://emma-stiefel.medium.com/plugging-holes-in-kaggles-titanic-dataset-an-introduction-to-combining-datasets-with-fuzzywuzzy-60a686699da7\n",
     "license": "Public",
     "citation": "The principal source for data about Titanic passengers is the Encyclopedia Titanica (http://www.encyclopedia-titanica.org/). The datasets used here were begun by a variety of researchers. One of the original sources is Eaton & Haas (1994) Titanic: Triumph and Tragedy, Patrick Stephens Ltd, which includes a passenger list created by many researchers and edited by Michael A. Findlay.\n\nThomas Cason of UVa has greatly updated and improved the titanic data frame using the Encyclopedia Titanica and created the dataset here. Some duplicate passengers have been dropped, many errors corrected, many missing ages filled in, and new variables created.\n",
-  
+
     "distribution": [
       {
-        "@type": "FileObject",
+        "@type": "sc:FileObject",
         "name": "passengers-table",
         "contentUrl": "https://www.openml.org/data/get_csv/16826755/phpMYEkMl",
         "contentSize": "117743 B",
         "sha256": "c617db2c7470716250f6f001be51304c76bcc8815527ab8bae734bdca0735737",
         "encodingFormat": "text/csv"
       }
     ],
-  
+
     "recordSet": [
+      {
+        "@type": "ml:RecordSet",
+        "name": "gender_enums",
+        "description": "Maps gender keys (0, 1) to labeled values.",
+        "key": "#{key}",
+        "field": [
+            { "name": "key", "@type": "ml:Field", "dataType": "sc:Integer" },
+            { "name": "label", "@type": "ml:Field", "dataType": "sc:String" },
+            {
+                "name": "url",
+                "@type": "ml:Field",
+                "dataType": ["sc:Url", "wd:48277"]
+            }
+        ],
+        "data": [
```

Comment:
`data` is a new field, right? I could not find a similar field in other schema.org types.
If it's not an existing field, would it make sense to be more descriptive here, such as `possible_value` instead of `data`? Or do we see other uses for this field?

Reply:
Excellent point. I completely agree with you on a conceptual level. 

On a practical level, I'm having some doubts on the complexity of this representation. We're re-using an existing mechanism (recordSet). If we were to use a dedicated mechanism for possible values, with "key", "label", and "url" (maybe "description"?) pre-defined, it would be easier to understand just by glancing at the `json`.

Just as a discussion point, I'm interested in your thoughts!

Reply:
That's a good point, I think, but I'm still having some doubts whether we really need this expressiveness.

For any enum, there will be a single field in the data, if I'm not mistaken. E.g. `capital=0` or `capital=1`. If `key=0` then  `label="France, Paris"`, if `key=1` then `label="Netherlands, Amsterdam"`. It's true that you lose some semantic information by just storing the enum's possibilities in a single label, but I'm not sure if this semantic information is worth the additional complexity.

Side thought: could we also map multiple fields to a single `RecordSet`? E.g. `cities` will have different possible values based on the `country` field? I'm not sure if we can handles these cases now (do we want to?)

In any case, I'm just playing the devils advocate here, I'm happy to go along with this solution.